### PR TITLE
Fix GetThreadName: it was always returning the current thread's name

### PR DESCRIPTION
**Pull request by dpallotti**
Use an implementation similar to the one previously used in `OrbitLinuxTracing`.
Also add the unit test that would catch this.

Bug: http://b/175663948

### CODE REVIEW

**Review by dimitry- — APPROVED** [review]

File: OrbitBase/ThreadUtilsLinux.cpp [hunk 1]
```diff
@@ -22,13 +23,16 @@ pid_t GetCurrentThreadId() {
...
-  if (result != 0) {
-    ERROR("Getting thread name for tid %d. Error %d", tid, result);
-    return {};
+  std::string comm_filename = absl::StrFormat("/proc/%d/comm", tid);
+  ErrorMessageOr<std::string> comm_content = ReadFileToString(comm_filename);
+  if (!comm_content.has_value()) {
+    ERROR("Getting thread name for tid %d: %s", tid, comm_content.error().message());
+    return "";
+  }
+  if (comm_content.value().back() == '\n') {
```
Comment (dimitry-) [inline OrbitBase/ThreadUtilsLinux.cpp:32]: absl::StripSuffix? It is slower though....

Comment (dimitry-) [reply OrbitBase/ThreadUtilsLinux.cpp:32]: also - what if it is empty.. also not sure if this could happen, can you set "" name for a thread?

Comment (dimitry-) [reply OrbitBase/ThreadUtilsLinux.cpp:32]: also this relies on the fact that value() returns a reference. maybe add auto value = std::move(comm_content.value()) and use that here and below?

File: OrbitBase/ThreadUtilsTest.cpp [hunk 2]
```diff
@@ -51,3 +52,43 @@ TEST(ThreadUtils, GetSetThreadNames) {
...
+TEST(ThreadUtils, GetThreadName) {
+  absl::Mutex mutex;
+#if _WIN32
+  using PidType = uint32_t;
+#else
+  using PidType = pid_t;
+#endif
+  PidType other_tid = 0;
+  bool other_name_read = false;
+  static const std::string kThreadName = "OtherThread";
```
Comment (dimitry-) [inline OrbitBase/ThreadUtilsTest.cpp:65]: add a test for empty name?

File: OrbitBase/ThreadUtilsLinux.cpp (hunk 1 above)
Comment (dpallotti) [reply OrbitBase/ThreadUtilsLinux.cpp:32]: I don't mind it, but indeed it copies the string. Up to you.

Comment (dimitry-) [reply OrbitBase/ThreadUtilsLinux.cpp:32]: > I don't mind it, but indeed it copies the string. Up to you.

I would keep it as it is. the '\n' check...

File: OrbitBase/ThreadUtilsTest.cpp (hunk 2 above)
Comment (dpallotti) [reply OrbitBase/ThreadUtilsTest.cpp:75]: Added for the current thread. Also rearranged the tests a bit.

File: OrbitBase/ThreadUtilsLinux.cpp (hunk 1 above)
Comment (dpallotti) [reply OrbitBase/ThreadUtilsLinux.cpp:33]: > also this relies on the fact that value() returns a reference. maybe add auto value = std::move(comm_content.value()) and use that here and below?

No, it's normal that `value` returns a reference, it would be quite a weird interface otherwise.

Comment (dpallotti) [reply OrbitBase/ThreadUtilsLinux.cpp:33]: > also - what if it is empty.. also not sure if this could happen, can you set "" name for a thread?

Done.

Comment (dimitry-) [reply OrbitBase/ThreadUtilsLinux.cpp:33]: I would still add a check for value.empty(), I am not sure if the test works because comm always has \n at the end or we are just lucky.

**Review by dimitry- — APPROVED** [review]

File: OrbitBase/ThreadUtilsLinux.cpp (hunk 1 above)
Comment (dpallotti) [reply OrbitBase/ThreadUtilsLinux.cpp]: Aaah, you are right, I didn't understand what you meant above, this worked because the `\n` was always there but if we have the `if` we need to do it right! Added `!comm_content.value().empty() &&` to the condition.

Comment (dpallotti) [reply OrbitBase/ThreadUtilsLinux.cpp]: Tell me if it looks good now before I get it wrong again.

**Review by dimitry- — APPROVED** [review]
lgtm!
